app/controllers/api/connect/v3/subscriptions/systems_controller.rb

```diff
@@ -1,16 +1,19 @@
 class Api::Connect::V3::Subscriptions::SystemsController < Api::Connect::BaseController
 
   def announce_system
-    @system = System.create!(hostname: params[:hostname], hw_info: HwInfo.new(hw_info_params))
+    @system = System.create!(hostname: params[:hostname], system_information: hwinfo_params[:hwinfo].to_json)
 
     logger.info("System '#{@system.hostname}' announced")
     respond_with(@system, serializer: ::V3::SystemSerializer, location: nil)
   end
 
   private
 
-  def hw_info_params
-    return {} if params[:hwinfo].blank?
-    params[:hwinfo].permit(:cpus, :sockets, :arch, :hypervisor, :uuid, :cloud_provider)
+  def hwinfo_params
+    # Allow all attributes without validating the key structure
+    # This is fine since the systems are only internal and RMT users
+    # can save in their own database what ever they want.
+    # When forwarded to SCC, SCC validates the payload for correctness.
```

Comment:
is that validation a big effort ? it may be worthy not to let it reach SCC
fine either way

Reply:
Dropping this is exactly the point of all the changes. Goal is to have RMT sent any payload to SCC. This removes the need to adapt RMT everytime when SUSEConnect is changing its payload. Less work for developers, better for us.

Reply:
And SCC does the validation anyway.